staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -110,8 +111,68 @@ func parsePorts(ports []string) ([]ForwardedPort, error) {
 	return forwards, nil
 }
 
+type listenAddress struct {
+	address  string
+	protocol string
+}
+
+func parseAddresses(addressesToParse []string) ([]listenAddress, error) {
+	var addresses []listenAddress
+	var al []string
+	// replace non IP entries with their resolved addresses or remove completely
+	// "localhost" will get replaced with 127.0.0.1 and ::1
+	for _, addr := range addressesToParse {
+		if addr == "localhost" {
+			al = append(al, "127.0.0.1")
+			al = append(al, "::1")
+		} else if net.ParseIP(addr) != nil {
+			al = append(al, addr)
+		} else {
+			var hosts []string
+			hosts, err := net.LookupHost(addr)
```

Comment:
I wouldn't expect to do DNS resolution internally... specifying the actual address (`127.0.0.1`, `::1`, `1.2.3.4`, `0.0.0.0`, etc) is consistent with all other bind address options in kubernetes commands

Reply:
I must have had a reason to include it, but after all this time I don't really remember. Probably just some minor convenience like not having to script IP4/6 retrieval when you know the name. I am not going to argue much in favor of it though, so if that is a problem, I'll just strip it away.

Reply:
(FTR, I interpret @liggitt's comment as meaning "don't accept localhost as an option, so that resolution isn't necessary".)